### DISCUSSION TITLE
Cannot read property 'match' of null: wire-directives.js (in function parseOutMethodAndParams)

### DISCUSSION THREAD

**Discussion by michabbb** [livewire/livewire#3028] [7. Bugs (archived)]
### Description

okay, i try: i have a livewire component inside my page:

```html
<livewire:show-list/>
````

inside this component I have a list of items:

```html
@foreach($list as $item)
<div onclick="document.getElementById('{{'info_circle_'.$item->name}}').classList.add('animate-pulse');" wire:click="trigger('{{$item->name}}')">
   @if($active===$item->name)
      <button wire:click.prevent="findDates('{{$item->itemnr}}')" wire:loading.class="animate-pulse">Find</button>
      @livewire('show-dates', ['dates' => $dates,'itemnr' => $item->itemnr], key($item->itemnr))
   @endif
@endforech
```

the "trigger" function here sets `$active` = `$item->name` so that the new Button "Find" appears.
as soon as I click on "findDates" i get the error:

```
wire-directives.js:86 Uncaught TypeError: Cannot read property 'match' of null
    at Directive.parseOutMethodAndParams (wire-directives.js:86)
    at Directive.get (wire-directives.js:58)
    at node_initializer.js:204
    at Component.value (index.js:564)
    at HTMLDivElement.<anonymous> (node_initializer.js:196)
```

because:

![image](https://user-images.githubusercontent.com/3524595/116729718-cace7680-a9e7-11eb-8e01-c711913c3977.png)

`rawMethod` is empty.

my structure here is like a wizard, where a livewire component is inside a component that has a component inside, as further you go the more components are nested within each other:

- you get a list (top component)
- you click on a item (something gets loaded for that item)
- you choose again from the previews loaded data....
- and so on... 

like a wizard. i hope this is understandable in some way ;)

**AND**: even that error occurs, my script/site is working fine, everything gets loaded.

### Exact steps to reproduce

i don´t know, I hope my infos are helpful 🤷‍♂️ 

**Context**
- Livewire version: v2.3.17
- Laravel version: v8.9.0
- Alpine version: no idea
- Browser: Chrome (Canary)

**Comment by calebporzio** [comment]
Can you provide a repository to easily recreate the issue? Then others can more easily help you. I say this because my guess is this will only occur after nesting components

**Comment by supersophisticated22** [comment]
Same problem here with a for each loop and list item element After many hours I found a dirty fix. I added an empty list item before the for each. Now works without a problem

**Comment by joshhanley** [comment]
@supersophisticated22 or @michabbb can either of you provide a repo that demonstrates the issue so we can investigate it?

Thanks! 🙂

**Comment by michabbb** [comment]
i try, i am quite busy at the moment, maybe this weekend....

**Comment by haviladev** [comment]
Today I had a similar issue and got fixed adding the prevent to the all wire click events, `wire:click.prevent` 
(don't ask me why)

**Comment by sneakylenny** [comment]
## Fix: DOM diffing issues
I ran into this issue as well but after finding nothing useful or something that worked in this discussion. I went to the almighty docs and found that in **my case** I was suffering from "[DOM diffing issues](https://laravel-livewire.com/docs/2.x/troubleshooting#dom-diffing-issues)" which can be fixed by applying the "[cure](https://laravel-livewire.com/docs/2.x/troubleshooting#dom-diffing-cures)". Which is adding unique `wire:key` props to the cursed elements.

Hope this helps someone since this discussion will pop up when looking for this error.

**Reply by raffaele-capasso** [reply]
@timyourivh straight to the point.. Thanks! 🙏🏽
